### PR TITLE
Add tests for upload request parameters demonstrating bug

### PR DESCRIPTION
**Pull request by YomesInc**
❗ This PR should not be merged as-is. It adds a couple of tests (including a breaking one) to demonstrate a potential bug.

### Bug description:

The [documentation](https://cloudinary.com/documentation/image_upload_api_reference#upload_optional_parameters) for passing `headers` in the Upload API's `options` states that it accepts a string or possibly an array of strings.

It even gives an example of a possible value: `X-Robots-Tag: noindex`.

The SDK however only accepts the headers as arrays and not as a string.

In other words, this works: 
````php
$api->upload($file, ['headers' => ['Link' => '1', 'X-Robots-Tag' => 'noindex']]);
````

This doesn't work:
````php
$api->upload($file, ['headers' => ['Link: 1', 'X-Robots-Tag: noindex']]);
````

---

Note that the test for [ApiUtils::serializeHeaders()](https://github.com/cloudinary/cloudinary_php/blob/d70503c0f8d627c573171c63c2d6ad64a84d1ab2/tests/Unit/Utils/ApiUtilsTest.php#L67) demonstrates exactly how both **should** be supported.

### CODE REVIEW

**Comment by const-cloudinary** [comment]
@YomesInc , yes, it looks like a bug, but not in place where the test checks it. 
The field `headers` is a request body parameter that is passed to the Cloudinary server. The issue here, is that it is also passed to the `ApiClient` and it treats it as request options and also consumes it. We need to find a way to both pass `headers` request parameter to the API, and allow customers to configure custom headers in the `ApiClient` itself.
